Sources/AppCoinsSDK/Domain/Repositories/Transaction/TransactionRepository.swift

```diff
@@ -85,9 +86,13 @@ internal class TransactionRepository: TransactionRepositoryProtocol {
                         }
                     }
                 } else if ["INVALID_TRANSACTION", "FAILED", "CANCELED", "FRAUD", "UNKNOWN"].contains(transactionRaw.status) {
+                    
+                    self.AnalyticsService.recordPaymentStatus(status: transactionRaw.status)
                     // Deal with different types of errors
                     completion(.failure(.failed()))
                 } else if transactionRaw.status == "COMPLETED" {
+                    
+                    self.AnalyticsService.recordPaymentStatus(status: transactionRaw.status)
```

Comment:
I know it's against clean architecture practices but use AnalyticsUseCases here anyway